File: PYTHON/PROGRAM/Memory.py
from Ram import Ram
from Harddrive import Harddrive

class Memory(Ram, Harddrive):
    def __init__(self, kapasitas_ram=0, ddr="", kapasitas_storage=0, tipe_drive="", 
                 merk="", nama="", jenis_memory="", kecepatan_transfer=0):
        # Memanggil constructor dari kedua parent class
        Ram.__init__(self, kapasitas_ram, ddr, merk, nama)
        Harddrive.__init__(self, kapasitas_storage, tipe_drive, merk, nama)
        self.__jenis_memory = jenis_memory
        self.__kecepatan_transfer = kecepatan_transfer
    
    #Setter
    def set_jenis_memory(self, jenis):
        self.__jenis_memory = jenis
    
    def set_kecepatan_transfer(self, kecepatan):
        self.__kecepatan_transfer = kecepatan
    
    #Getter
    def get_jenis_memory(self):
        return self.__jenis_memory
    
    def get_kecepatan_transfer(self):
        return self.__kecepatan_transfer
    
    # Method tambahan
    def optimize(self):
        print(f"Mengoptimalkan kinerja memory {self.get_merk()} {self.get_nama()}")
        print(f"Meningkatkan kecepatan transfer dari {self.__kecepatan_transfer} "
              f"MB/s menjadi {self.__kecepatan_transfer * 1.2} MB/s")

File: JAVA/PROGRAM/Memory.java
public class Memory extends Ram {
    private int kapasitasStorage;
    private String tipeDrive;
    private String jenisMemory;
    private int kecepatanTransfer;
    
    // Konstruktor
    public Memory() {
        super();
        this.kapasitasStorage = 0;
        this.tipeDrive = "";
        this.jenisMemory = "";
        this.kecepatanTransfer = 0;
    }
    
    public Memory(int kapasitasRam, String ddr, int kapasitasStorage, String tipeDrive, 
                  String merk, String nama, String jenisMemory, int kecepatanTransfer) {
        super(kapasitasRam, ddr, merk, nama);
        this.kapasitasStorage = kapasitasStorage;
        this.tipeDrive = tipeDrive;
        this.jenisMemory = jenisMemory;
        this.kecepatanTransfer = kecepatanTransfer;
    }
    
    // Getter dan Setter untuk attributes tambahan
    public int getKapasitasStorage() {
        return kapasitasStorage;
    }
    
    public void setKapasitasStorage(int kapasitasStorage) {
        this.kapasitasStorage = kapasitasStorage;
    }
    
    public String getTipeDrive() {
        return tipeDrive;
    }
    
    public void setTipeDrive(String tipeDrive) {
        this.tipeDrive = tipeDrive;
    }
    
    public String getJenisMemory() {
        return jenisMemory;
    }
    
    public void setJenisMemory(String jenisMemory) {
        this.jenisMemory = jenisMemory;
    }
    
    public int getKecepatanTransfer() {
        return kecepatanTransfer;
    }
    
    public void setKecepatanTransfer(int kecepatanTransfer) {
        this.kecepatanTransfer = kecepatanTransfer;
    }
    
    // Method tambahan
    public void optimize() {
        System.out.println("Mengoptimalkan kinerja memory " + this.getMerk() + " " + this.getNama());
        System.out.println("Meningkatkan kecepatan transfer dari " + this.kecepatanTransfer + 
                           " MB/s menjadi " + (this.kecepatanTransfer * 1.2) + " MB/s");
    }
}